src/ui/QuerySummary/QuerySummary.ts

```diff
@@ -86,58 +89,108 @@ export class QuerySummary extends Component {
+    } else if (queryResults.results.length == 0) {
       this.displayInfoOnNoResults();
     } else {
       this.lastKnownGoodState = this.queryStateModel.getAttributes();
     }
   }
 
+  private handleQuerySuccess(data: IQuerySuccessEventArgs) {
+    Assert.exists(data);
+    this.render(data.query, data.results);
+  }
+
+  private isInfiniteScrollingMode() {
+    const allResultsLists = $$(this.root).findAll(`.${Component.computeCssClassNameForType('ResultList')}`);
+    const anyResultListIsUsingInfiniteScroll = any(allResultsLists, resultList => {
+      return (get(resultList) as ResultListModule.ResultList).options.enableInfiniteScroll;
+    });
+    return anyResultListIsUsingInfiniteScroll;
+  }
+
+  private formatSummary(queryPerformed: IQuery, queryResults: IQueryResults) {
+    const first = Globalize.format(queryPerformed.firstResult + 1, 'n0');
+    const last = Globalize.format(queryPerformed.firstResult + queryResults.results.length, 'n0');
+    const totalCount = Globalize.format(queryResults.totalCountFiltered, 'n0');
+    const query = queryPerformed.q ? escape(queryPerformed.q.trim()) : '';
+
+    const highlightFirst = $$('span', { className: 'coveo-highlight' }, first).el;
+    const highlightLast = $$('span', { className: 'coveo-highlight' }, last).el;
+    const highlightTotal = $$('span', { className: 'coveo-highlight' }, totalCount).el;
+    const highlightQuery = $$('span', { className: 'coveo-highlight' }, query).el;
+
+    return {
+      first,
+      last,
+      totalCount,
+      query,
+      highlightFirst,
+      highlightLast,
+      highlightTotal,
+      highlightQuery
+    };
+  }
+
+  private renderSummaryInStandardMode(queryPerformed: IQuery, queryResults: IQueryResults) {
+    if (queryResults.results.length > 0) {
+      const { query, highlightFirst, highlightLast, highlightTotal, highlightQuery } = this.formatSummary(queryPerformed, queryResults);
+
+      if (query) {
+        this.textContainer.innerHTML = l(
+          'ShowingResultsOfWithQuery',
+          highlightFirst.outerHTML,
+          highlightLast.outerHTML,
+          highlightTotal.outerHTML,
+          highlightQuery.outerHTML,
+          queryResults.results.length
+        );
+      } else {
+        this.textContainer.innerHTML = l(
+          'ShowingResultsOf',
+          highlightFirst.outerHTML,
+          highlightLast.outerHTML,
+          highlightTotal.outerHTML,
+          queryResults.results.length
+        );
+      }
+    }
+  }
+
+  private renderSummaryInInfiniteScrollingMode(queryPerformed: IQuery, queryResults: IQueryResults) {
+    if (queryResults.results.length > 0) {
+      const { query, highlightQuery, highlightTotal } = this.formatSummary(queryPerformed, queryResults);
+
+      if (query) {
+        this.textContainer.innerHTML = l(
+          'ShowingResultsWithQuery',
+          highlightTotal.outerHTML,
+          highlightQuery.outerHTML,
+          queryResults.results.length
```

Comment:
Are we missing the `query` parameter here?

Reply:
No, we want to use the "highlightQuery" variable, which contains HTML for highlighting the query.